### PR TITLE
gh-109286: Updated macOS SQLite installer to 3.43.1

### PR DESCRIPTION
**Pull request by ghost**
I updated the macOS SQLite installer, all changes were made in the build-installer.py file. I'll be doing the Windows installer next in a separate PR as requested.

* Issue: gh-109286

### CODE REVIEW

File: Mac/BuildScript/build-installer.py [hunk 1]
```diff
@@ -359,16 +359,16 @@ def library_recipes():
...
+              name="SQLite 3.43.1",
+              url="https://sqlite.org/2023/sqlite-autoconf-3430100.tar.gz",
+              checksum="77e61befe9c3298da0504f87772a24b0",
+	          extra_cflags=('-Os '
+                '-DSQLITE_ENABLE_FTS5 '
+                '-DSQLITE_ENABLE_FTS4 '
+                '-DSQLITE_ENABLE_FTS3_PARENTHESIS '
+                '-DSQLITE_ENABLE_RTREE '
+                '-DSQLITE_OMIT_AUTOINIT '
+                '-DSQLITE_TCL=0 '
```
Comment (erlend-aasland) [inline Mac/BuildScript/build-installer.py]: Please revert these style changes. _Only change line 362, 363, and 364._

See the following PRs for earlier SQLite version bumps; **your PR should look very similar to these**:

- gh-104624
- gh-102998
- gh-98639

File: Misc/NEWS.d/next/macOS/2023-10-04-23-38-24.gh-issue-109286.1ZLMaq.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Upgraded macOS SQLite installer to 3.43.1
```
Comment (erlend-aasland) [inline Misc/NEWS.d/next/macOS/2023-10-04-23-38-24.gh-issue-109286.1ZLMaq.rst]:
```suggestion
Update macOS installer to use SQLite 3.43.1.
```

File: Mac/BuildScript/build-installer.py (hunk 1 above)
Comment (erlend-aasland) [reply Mac/BuildScript/build-installer.py]:
```suggestion
              extra_cflags=('-Os '
                            '-DSQLITE_ENABLE_FTS5 '
                            '-DSQLITE_ENABLE_FTS4 '
                            '-DSQLITE_ENABLE_FTS3_PARENTHESIS '
                            '-DSQLITE_ENABLE_RTREE '
                            '-DSQLITE_OMIT_AUTOINIT '
                            '-DSQLITE_TCL=0 '
```

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
> I updated the macOS SQLite installer, all changes were made in the build-installer.py file. I'll be doing the Windows installer next in a separate PR as requested.

Before doing the Windows PRs, please take a close look at the earlier Windows PRs; your PR should look very similar to these:

- gh-104625
- gh-102999

**Comment by jtranquilli** [comment]
> > I updated the macOS SQLite installer, all changes were made in the build-installer.py file. I'll be doing the Windows installer next in a separate PR as requested.
> 
> _Before_ doing the Windows PRs, please take a close look at the earlier Windows PRs; your PR should look **very similar to these**:
> 
> * [gh-104623: Update Windows installer to use SQLite 3.42.0 #104625](https://github.com/python/cpython/pull/104625)
> * [gh-102997: Update Windows installer to SQLite 3.41.2. #102999](https://github.com/python/cpython/pull/102999)

Sounds good, sorry about that. I had messed around with those lines because I was getting a trailing white space error, and then in my previous pr someone had asked me to realign the indentation of extra_cflags, so I thought it was accidentally over-indented. Just out of curiosity, why is the extra indentation preferred in that context?

**Comment by erlend-aasland** [comment]
Why was this closed?

**Comment by erlend-aasland** [comment]
> Just out of curiosity, why is the extra indentation preferred in that context?

It's not that the extra indentation is preferred; the point is that any change should be as isolated as possible. Any style changes (or other unrelated changes) should be left out.

**Comment by jtranquilli** [comment]
> Why was this closed?

It was an accident. I was asked to separate the changes for macOS and Windows into two separate PRs, and despite using different branches for each set of changes, GitHub was not allowing me to create separate PRs for each, so I submitted the PR with the macOS changes, then deleted that branch on my GitHub page so that the changes made in that branch would not be included in my PR for the Windows changes, then submitted the PR for the Windows changes. I didn't realize that deleting the branch on my personal page would close this PR and that was the only work around I could think of to isolate the changes for the two operating systems since Github was seemingly not allowing me to isolate the changes made in each branch into two separate PRs. If you merge the PR I made for the Windows SQLite installer upgrade I'll submit a new PR for the macOS installer upgrades, but other than deleting the branch locally I'm not sure how I can isolate those changes as requested. Sorry for the hassle.

**Comment by erlend-aasland** [comment]
Can you please reopen this PR?

**Comment by jtranquilli** [comment]
> Can you please reopen this PR?

I'm prevented from doing so because the branch was deleted, re-uploading it doesn't seem to help either. All the requested changes were saved on my local branch though so I'll just create a new pr. Again, sorry for the hassle.

**Pull request closed without merging**